File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_100438/typescript/003_fibonacci/solution.ts
const N = parseInt(process.argv[2]);

if (N < 1) {
  process.exit(0);
}

let a: number = 1;
let b: number = 1;

const result: number[] = [];

result.push(a);
result.push(b);

while (b <= N) {
  const next = a + b;
  a = b;
  b = next;
  if (b <= N) {
    result.push(b);
  } else {
    break;
  }
}

for (const num of result) {
  console.log(num);
}

File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_100438/python/003_fibonacci/solution.py
import sys

if len(sys.argv) != 2 or not sys.argv[1].isdigit():
    print("Usage: python fibonacci.py <number>")
    sys.exit(1)

n = int(sys.argv[1])

if n < 1:
    pass  # No output for negative or zero N
else:
    a, b = 1, 1
    fib_numbers = [a] if n >= 0 else []
    while b <= n:
        fib_numbers.append(b)
        a, b = b, a + b
        if b > n:
            break
    
    for num in fib_numbers:
        print(num)